tests/framework/extensions/provisioning/creates.go

```diff
@@ -114,8 +114,8 @@ func CreateProvisioningCluster(client *rancher.Client, provider Provider, cluste
 	return createdCluster, err
 }
 
-// CreateProvisioningCustomCluster provisions a non-rke1 cluster using a 3rd party client for its nodes, then runs verify checks
-func CreateProvisioningCustomCluster(client *rancher.Client, externalNodeProvider ExternalNodeProvider, clustersConfig *clusters.ClusterConfig) (*v1.SteveAPIObject, error) {
+// CreateProvisioningRKE2CustomCluster provisions a rke2 cluster using a 3rd party client for its nodes, then runs verify checks
```

Comment:
this is the opposite side of my argument in an earlier function; too much is shared here between rke2 and k3s. I argue this should remain 1 function for rke2/k3s.

Reply:
Yeah, the problem here is that RKE2 and K3S custom clusters call different hardening functions. So then we would have to completely rework the `hardening` extensions folder